packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/platform_navigation_delegate.dart

```diff
@@ -9,6 +9,19 @@ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
 import 'webview_platform.dart';
 
+/// Signature for callbacks that report a pending navigation request.
+typedef NavigationRequestCallback = FutureOr<bool> Function(
+    {required String url, required bool isForMainFrame});
```

Comment:
Could you change this to a single class with both of these as fields? e.g.

```dart
class NavigationRequest {
  final String url;
  final bool isMainFrame;
}
```

I realized that platforms may want to pass back additional details like with `WebResourceError`.